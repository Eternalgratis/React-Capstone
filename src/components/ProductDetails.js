import React, { useEffect } from 'react';
import { useParams, Link } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { fetchProducts } from '../redux/productApi';
import { getProductDetails } from '../redux/productDetails';
import './productDetails.css';

const ProductDetails = () => {
  const { id } = useParams();
  const { product } = useSelector((state) => state.productDetail);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchProducts())
      .unwrap()
      .then(((result) => {
        dispatch(getProductDetails({ products: result, id: Number(id) }));
      }));
  }, []);

  const unit = '$';
  return (
    <div id={product.id}>
      <Link to="/">
        <button type="button">Go back</button>
      </Link>
      <div className="details-container">
        <img
          src={product.image}
          alt={product.title}
          style={{ width: '600px', height: '800px' }}
        />
        <div className="sub-cont">
          <h1 className="head">{product.title}</h1>
          <p className="text-1">{product.description}</p>
          <p className="text">{`${unit} ${product.price}`}</p>
          <span className="cart"> Add to cart </span>
        </div>
      </div>
    </div>
  );
};

export default ProductDetails;
