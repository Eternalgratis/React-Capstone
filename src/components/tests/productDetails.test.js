import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/configureStore';
import ProductDetails from '../ProductDetails';

describe('to test the productDetails components', () => {
  test('matches the productDetails DOM snapshot', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <ProductDetails />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
