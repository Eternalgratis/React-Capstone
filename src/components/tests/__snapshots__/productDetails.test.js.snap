// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to test the productDetails components matches the productDetails DOM snapshot 1`] = `
<div>
  <a
    href="/"
    onClick={[Function]}
  >
    <button>
      Go back
    </button>
  </a>
  <div
    className="details-container"
  >
    <img
      style={
        Object {
          "height": "800px",
          "width": "400px",
        }
      }
    />
    <div
      className="sub-cont"
    >
      <h1
        className="head"
      />
      <p
        className="text-1"
      />
      <p
        className="text"
      >
        $ undefined
      </p>
      <span
        className="cart"
      >
         Add to cart 
      </span>
    </div>
  </div>
</div>
`;
